src/client/dfs/dfs.c

```diff
@@ -4575,10 +4584,10 @@ dfs_chmod(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode)
 	rc = fetch_entry(dfs->layout_v, oh, DAOS_TX_NONE, name, len, true, &exists, &entry,
 			 0, NULL, NULL, NULL);
 	if (rc)
-		D_GOTO(out, rc);
+		return rc;
```

Comment:
Do we still need to run the following code here instead of returning immediately?

out:
	if (S_ISLNK(entry.mode)) {
		dfs_release(sym);
		daos_obj_close(oh, NULL);
	}

Reply:
no. it was a bug actually to run it if we error here.